pkg/manager/memory/actionstatus.go

```diff
@@ -0,0 +1,61 @@
+package memory
+
+import (
+	"context"
+	"sync"
+
+	"github.com/puppetlabs/relay-core/pkg/model"
+)
+
+type ActionStatusKey struct {
+	ActionHash string
+}
+
+type ActionStatusMap struct {
+	mut    sync.RWMutex
+	status map[ActionStatusKey]*model.ActionStatus
+}
+
+func (s *ActionStatusMap) Get(key ActionStatusKey) (*model.ActionStatus, error) {
+	s.mut.Lock()
+	defer s.mut.Unlock()
+
+	return s.status[key], nil
+}
+
+func (s *ActionStatusMap) Set(key ActionStatusKey, message *model.ActionStatus) {
+	s.mut.Lock()
+	defer s.mut.Unlock()
+
+	s.status[key] = message
+}
+
+func NewActionStatusMap() *ActionStatusMap {
+	return &ActionStatusMap{
+		mut:    sync.RWMutex{},
+		status: make(map[ActionStatusKey]*model.ActionStatus),
+	}
+}
+
+type ActionStatusManager struct {
+	me model.Action
+	m  *ActionStatusMap
+}
+
+func (s *ActionStatusManager) Get(ctx context.Context, action model.Action) (*model.ActionStatus, error) {
+	return s.m.Get(ActionStatusKey{ActionHash: action.Hash().String()})
+}
+
+func (s *ActionStatusManager) Set(ctx context.Context, sm *model.ActionStatus) error {
+	s.m.Set(ActionStatusKey{ActionHash: s.me.Hash().String()}, sm)
+	return nil
+}
+
+var _ model.ActionStatusManager = &ActionStatusManager{}
+
+func NewActionStatusManager(step *model.Step, backend *ActionStatusMap) *ActionStatusManager {
```

Comment:
I think this could take any action, not just a step, right?

Reply:
Yeah, it originally only supported steps to have parity with the ConfigMap implementation. But I removed it at the last moment; I keep going back and forth on whether it should be similarly restricted or not. Considering we're making the general design to be viable for all actions.

Reply:
Ahh, and I meant the in-memory `Set` which is currently not restricting the Action type, as opposed to the ConfigMap implementation. I didn't even notice that the `step *model.Step` still needs to be updated, thanks.